sim-validator-assignment/src/run.rs

```diff
@@ -40,15 +43,24 @@ pub fn run(config: &Config) -> anyhow::Result<()> {
     let mut num_corrupted_shards = 0;
 
     for block_height in 0..config.num_blocks {
-        let mut ordered_seats = new_ordered_seats(&validators);
-        let shuffled_seats = ShuffledSeats::new(&mut ordered_seats);
+        let mut seats = new_ordered_seats(&validators);
+        let shuffled_seats = ShuffledSeats::new(&mut seats);
+
+        let mut partial_seats = config
+            .include_partial_seats
+            .then(|| new_ordered_partial_seats(&validators, config.stake_per_seat));
```

Comment:
I think the code following this assignment could be simplified if instead of assigning `partial_seats` as an `Option` instead we simply make it empty if partial seats are not enabled

```rust
let mut partial_seats = if (config.include_partial_seats) {
    new_ordered_partial_seats(&validators, config.stake_per_seat)
} else {
    Vec::new()
};
```

The rest of the code needs to handle the case where `partial_seats` is empty regardless because it is possible for partial seats to be enabled in the config but all validators have stake which evenly divides the seat size.

Reply:
I considered both ways and initially went with `Option<PartialSeats>` since it can represent the difference between not having partial seats due to:

- configuration (partial seats disabled)
- alll validators’ stake is zero or fills full seats without remainder

Though I don’t expect that to matter for the simulation in places that currently use `Option<PartialSeats>`.  So I agree that it makes sense to keep it simple and discard the `Option`.

Changed in 0b64192.